packages/auth-construct/src/construct.ts

```diff
@@ -893,40 +939,41 @@ export class AmplifyAuth
         output.mfaTypes = JSON.stringify(mfaTypes);
       }
     }
-    const oauthMappings = this.providerSetupResult.oauthMappings;
-    if (oauthMappings[authProvidersList.amazon]) {
-      output.amazonClientId = oauthMappings[authProvidersList.amazon];
+    const oAuthMappings = this.providerSetupResult.oAuthMappings;
+    if (oAuthMappings[authProvidersList.amazon]) {
+      output.amazonClientId = oAuthMappings[authProvidersList.amazon];
     }
-    if (oauthMappings[authProvidersList.facebook]) {
-      output.facebookClientId = oauthMappings[authProvidersList.facebook];
+    if (oAuthMappings[authProvidersList.facebook]) {
+      output.facebookClientId = oAuthMappings[authProvidersList.facebook];
     }
-    if (oauthMappings[authProvidersList.google]) {
-      output.googleClientId = oauthMappings[authProvidersList.google];
+    if (oAuthMappings[authProvidersList.google]) {
+      output.googleClientId = oAuthMappings[authProvidersList.google];
     }
-    if (oauthMappings[authProvidersList.apple]) {
-      output.appleClientId = oauthMappings[authProvidersList.apple];
+    if (oAuthMappings[authProvidersList.apple]) {
+      output.appleClientId = oAuthMappings[authProvidersList.apple];
```

Comment:
How about...?

```suggestion
    const oAuthMappings = this.providerSetupResult.oAuthMappings;
    Object.entries(authProvidersList).forEach(([authProvider, endpoint]) => {
        if (oauthMappings[endpoint]) {
        output[`${authProvider}ClientId` as keyof AuthOutput['payload']] = oauthMappings[endpoint];
        }
    })
```

Reply:
Very clean suggestion , I'll include this in the refactor PR with Praveens other suggestions if thats OK